tests/unit/test_session.py

```diff
@@ -367,6 +367,15 @@ def test_credential_provider_not_called_when_creds_provided(self):
                          "explicit credentials were provided to the "
                          "create_client call.")
 
+    @mock.patch('botocore.client.ClientCreator')
+    def test_config_passed_to_client_creator(self, client_creator):
+        config = mock.Mock()
```

Comment:
We've discussed in the past not mocking value objects.  Let's just use the actual Config object here.